src/net_processing.cpp

```diff
@@ -2139,8 +2140,16 @@ bool ProcessMessages(CNode *pfrom)
 
                     // check if we should process the message.
                     CNode *pnode = noderef.get();
-                    if (pnode->fDisconnect || pnode->nSendSize > SendBufferSize())
+                    if (pnode->fDisconnect)
                     {
+                        // if the node is to be disconnected dont bother responding
+                        continue;
+                    }
+                    if (pnode->nSendSize > SendBufferSize())
+                    {
+                        // if the nodes send is full, delay the processing of this message until a time when
+                        // send is not full
+                        vPriorityRecvQ_delay.emplace_back(std::move(noderef), std::move(msg));
```

Comment:
doesn't this open us up to a memory exhaustion attack?  couldn't someone just flood us with priority messages which would then be stored in mem?

Reply:
In theory, i suppose it is possible. The attacking node would need to send us more messages than we can respond to in a given time span in order for that attack to be executed though. 

A mitigation would be to cap the amount of backlogged priority messages we have from any given peer to some amount of MB.